lib/poller.js

```diff
@@ -12,7 +13,9 @@ exports.do = (uuid, accessToken, apiUrl, pollStep = 1000, timeout = 30000) => {
   const timeoutPromise = new Promise((resolve, reject) => {
     timeoutID = setTimeout(() => {
       clearActions()
-      reject(new Error(`Timed out in ${timeout} ms.`))
+      // eslint-disable-next-line prefer-promise-reject-errors
+      reject(`User-specified or default timeout after ${helper.elapsedFmt(timeout)}.\n` +
+               'Analysis continues on and may complete on the server, so run again?')
```

Comment:
Why not throw new Error? 
```
const errorMessage = `User-specified or default timeout after ${helper.elapsedFmt(timeout)}.\nAnalysis continues on and may complete on the server, so run again?`
reject(new Error(errorMessage))
```
This way it is still possible to debug what caused the error and if you want nice human readable message you can do access `message` property.
Example:
```
try {
  throw new Error(errorMessage)
} catch(err) {
  console.log(err.message) // prints "User-specified or default timeout after ${helper.elapsedFmt(timeout)}.\nAnalysis continues on and may complete on the server, so run again?"
}
```

Reply:
I chose that because I don't want a stack trace in this particular situation. 

So the question is: how do we indicate that we don't want a stack trace to be shown?

Reply:
When  request fails due to timeout you reject results which is somewhat equivalent to Error (IMO). I believe it's better to reject with an error, not the result. 
On the client side we can write `console.log(err.message)` instead of `console.log(err)`
If code rejects with a string value then we have to add additional checks on the client side too tto knows which type of data is in reject. It will look similar to `typeof err === 'string' ? console.log(err) : console.log(err.message)` or `console.log(err.message || err)`

Reply:
I agree with @daniyarchambylov here, we should reject an error and handle it in the consumer code.

Reply:
@daniyarchambylov and @fgimenez Would one of you create a complete PR for this so we can evaluate the complete code?

It feels to me you two for the sake of convention rather than what this means. A timeout is not an error. 

I find it weird that those of you who go through great lengths to add promisfy and and await just to save a couple of microseconds in a stat() call have no problem in the overhead of creating an object populated with a traceback when that is not desired. 

Again, _a timeout is not an error_. Not all things that you throw have to be errors. 

That said, since there are two you who feel this way, I'll acquiese. But let's see the code so I fully understand what you are talking about.

Reply:
> If code rejects with a string value then we have to add additional checks on the client side too > to knows which type of data is in reject. It will look similar to `typeof err === 'string' ? console.log(err) : console.log(err.message)` or `console.log(err.message || err)`

But this is what happens when  ometimes we want a traceback or additional information that might be in `err` but sometimes we don't.   And that I believe is the situation here.

Reply:
Due to running out of time on this for a release, and lacking sufficient answers to my concern I am leaving this as is. After release we can come back and fix up in a mutally agreeable way.

The two concerns were: 

* having whether or not to include a stack decided at the point where the warning is raised, not the handler
* ensuring we don't print a stack in the handler for those messages where it doesn't make sense, which is known user errors. Printing a stack does make sense when the error really is something we hadn't anticipated and then occurs.